File: src/main/java/com/summer/lijiahao/utils/openapi/actions/FormatRequestJsonAction.java
package com.summer.lijiahao.utils.openapi.actions;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;
import com.alibaba.fastjson.serializer.SerializerFeature;
import com.intellij.icons.AllIcons;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.DumbAwareAction;
import com.summer.lijiahao.utils.openapi.ui.OpenApiTool;
import org.apache.commons.lang.StringUtils;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class FormatRequestJsonAction extends DumbAwareAction {
    private final OpenApiTool openApiTool;

    public FormatRequestJsonAction(OpenApiTool openApiTool) {
        super("格式化", "格式化", AllIcons.Actions.PrettyPrint);
        this.openApiTool = openApiTool;
    }

    public void actionPerformed(@NotNull AnActionEvent e) {
        String requestJson = this.openApiTool.getComponent("sendText", JTextArea.class).getText();
        if(StringUtils.isBlank(requestJson)) {
            return;
        }
        JSONObject jsonObject = JSONObject.parseObject(requestJson);
        String responseJson =
                JSON.toJSONString(jsonObject, SerializerFeature.PrettyFormat, SerializerFeature.WriteMapNullValue,
                        SerializerFeature.WriteDateUseDateFormat);
        this.openApiTool.getComponent("sendText", JTextArea.class).setText(responseJson);
    }
}

File: src/main/kotlin/com/summer/lijiahao/nccdevtool/opanapi/action/FormatRequestJsonAction.kt
package com.summer.lijiahao.nccdevtool.opanapi.action

import com.alibaba.fastjson.JSON
import com.alibaba.fastjson.JSONObject
import com.alibaba.fastjson.serializer.SerializerFeature
import com.intellij.icons.AllIcons
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.project.DumbAwareAction
import com.summer.lijiahao.nccdevtool.opanapi.ui.OpenApiTool

class FormatRequestJsonAction(openApiTool: OpenApiTool) :
    DumbAwareAction("格式化", "格式化", AllIcons.Actions.PrettyPrint) {
    private val openApiTool: OpenApiTool

    init {
        this.openApiTool = openApiTool
    }

    override fun actionPerformed(e: AnActionEvent) {
        val requestJson: String? = openApiTool.sendText?.text
        if (requestJson.isNullOrBlank()) {
            return
        }
        val jsonObject = JSONObject.parseObject(requestJson)
        val responseJson = JSON.toJSONString(
            jsonObject, SerializerFeature.PrettyFormat, SerializerFeature.WriteMapNullValue,
            SerializerFeature.WriteDateUseDateFormat
        )
        openApiTool.sendText?.text = responseJson
    }
}